source/geod24/LocalRest.d

```diff
@@ -721,52 +1026,76 @@ public final class RemoteAPI (API) : API
             mixin(q{
                 override ReturnType!(ovrld) } ~ member ~ q{ (Parameters!ovrld params)
                 {
-                    // we are in the main thread
-                    if (scheduler is null)
-                    {
-                        scheduler = new LocalScheduler;
-                        is_main_thread = true;
-                    }
-
-                    // `geod24.concurrency.send/receive[Only]` is not `@safe` but
-                    // this overload needs to be
                     auto res = () @trusted {
                         auto serialized = ArgWrapper!(Parameters!ovrld)(params)
                             .serializeToJsonString();
 
-                        auto command = Command(C.thisTid(), scheduler.getNextResponseId(), ovrld.mangleof, serialized);
-                        C.send(this.childTid, command);
+                        Command command;
+                        Response res;
 
-                        // for the main thread, we run the "event loop" until
-                        // the request we're interested in receives a response.
-                        if (is_main_thread)
+                        // from Node to Node
+                        if ((thisLocalTransceiver !is null) && (thisLocalTransceiver.belongToNode))
                         {
+                            command = Command(thisLocalTransceiver, thisLocalWaitingManager.getNextResponseId(), ovrld.mangleof, serialized);
+                            this.childTransceiver.send(command);
+
+                            res = thisLocalWaitingManager.waitResponse(command.id, this.timeout);
+                        }
+
+                        // from Non-Node to Node
+                        else
+                        {
+                            if (thisScheduler is null)
+                                thisScheduler = new FiberScheduler();
+
+                            if (thisLocalWaitingManager is null)
+                                thisLocalWaitingManager = new LocalWaitingManager();
+
+                            if (thisLocalTransceiver is null)
+                                thisLocalTransceiver = new LocalTransceiver(false);
+
                             bool terminated = false;
-                            runTask(() {
+                            command = Command(thisLocalTransceiver, thisLocalWaitingManager.getNextResponseId(), ovrld.mangleof, serialized);
+
+                            thisScheduler.spawn({
+                                this.childTransceiver.send(command);
+                            });
+
+                            thisScheduler.spawn({
```

Comment:
We shouldn't need to start 3 tasks, one should be enough.

Reply:
The third is waiting. So the third cannot be modified because it has to be executed at the same time as the other. The first and second were not executed normally when combined. So I knew that all three had to be implemented at the same time.